### PR TITLE
Put title in the right place

### PR DESCRIPTION
**Pull request by jrjohnson**
On an svg title is an element not an attribute.

### CODE REVIEW

File: lib/build-time-components/fa-icon.js [hunk 1]
```diff
@@ -33,7 +32,10 @@ module.exports = class FaIconComponent extends BuildTimeComponent {
       'dataIcon:data-icon',
       'dataPrefix:data-prefix',
     ];
-    this.layout`<path d={{iconPath}} fill=currentColor></path>`;
+    this.layout`
+      <title>{{title}}</title>
```
Comment (robmadole) [inline lib/build-time-components/fa-icon.js]: For the build time icons, are we going to get `<title></title>` if no title is present?

File: tests/integration/components/fa-icon-test.js [hunk 2]
```diff
@@ -77,12 +77,22 @@ test('it optionally renders fa-spin class', function(assert){
...
+  assert.expect(2);
   const title = 'awesome is as awesome does';
   this.set('title', title);
   this.set('faCoffee', faCoffee);
 
   this.render(hbs`{{fa-icon icon=faCoffee title=title}}`);
 
-  assert.equal(this.$('svg').attr('title'), title, 'Should include fa-spin class')
+  assert.equal(this.$('svg title').length, 1, 'has title element');
+  assert.equal(this.$('svg title').text().trim(), title, 'title is correct');
```
Comment (robmadole) [inline tests/integration/components/fa-icon-test.js]: :+1:

File: tests/integration/components/fa-icon-test.js [hunk 3]
```diff
@@ -77,12 +77,22 @@ test('it optionally renders fa-spin class', function(assert){
...
+});
+
+test('no title attribute gives no title element', function (assert) {
+  assert.expect(1);
+  this.set('faCoffee', faCoffee);
+
+  this.render(hbs`{{fa-icon icon=faCoffee}}`);
+
+  assert.equal(this.$('svg title').length, 0, 'has not title element');
```
Comment (robmadole) [inline tests/integration/components/fa-icon-test.js]: Nice.

File: lib/build-time-components/fa-icon.js (hunk 1 above)
Comment (jrjohnson) [reply lib/build-time-components/fa-icon.js]: Yes. I tried everything I could think of to work around this. I decided it's better to have a blank title when you don't expect it than a missing title when you do.  I'm also reaching out to some folks for help. I can take this out of this PR though if you want.

Comment (robmadole) [reply lib/build-time-components/fa-icon.js]: Oh no, leave it in. I agree with you. Is this full-on Handlebars?

Does this work?

```
{{if title}}
  <title>{{title}}</title>
{{/if}}
```

Comment (robmadole) [reply lib/build-time-components/fa-icon.js]: Actually I think it's `{{#if}}`...been awhile since I've used it.

Comment (jrjohnson) [reply lib/build-time-components/fa-icon.js]: I thought that would work and based on the docs at https://github.com/cibernox/ember-ast-helpers#layout it should have.  Alas, it doesn't.

**Comment by robmadole** [comment]
:+1:

**Pull request merged**